doc/source/rllib/rllib-rlmodule.rst

```diff
@@ -0,0 +1,571 @@
+
+                return {"action_dist": torch.distributions.Categorical(logits=action_logits)}
+
+
+
+        class BCTorchMultiAgentModuleWithSharedEncoder(MultiAgentRLModule):
+            def __init__(self, config: MultiAgentRLModuleConfig) -> None:
+                super().__init__(config)
+
+            def build(self):
+
+                module_specs = self.config.modules
+                module_spec = next(iter(module_specs.values()))
+                global_dim = module_spec.observation_space["global"].shape[0]
+                hidden_dim = module_spec.model_config_dict["fcnet_hiddens"][0]
+                shared_encoder = nn.Sequential(
+                    nn.Linear(global_dim, hidden_dim),
+                    nn.ReLU(),
+                    nn.Linear(hidden_dim, hidden_dim),
+                )
+
+                rl_modules = {}
+                for module_id, module_spec in module_specs.items():
+                    rl_modules[module_id] = BCTorchRLModuleWithSharedGlobalEncoder(
+                        encoder=shared_encoder,
+                        local_dim=module_spec.observation_space["local"].shape[0],
+                        hidden_dim=hidden_dim,
+                        action_dim=module_spec.action_space.n,
+                    )
+
+                self._rl_modules = rl_modules
+
+To construct this custom multi-agent RL odule, we need to pass the class to the :py:class:`~ray.rllib.core.rl_module.marl_module.MultiAgentRLModuleSpec` constructor. We need to also pass the :py:class:`~ray.rllib.core.rl_module.rl_module.SingleAgentRLModuleSpec` for each agent. This is because RLlib needs to know the observation, action spaces, and model hyper-parameters for each agent.
+
+.. code-block:: python
+
+    import gymnasium as gym
+    from ray.rllib.core.rl_module.rl_module import SingleAgentRLModuleSpec
+    from ray.rllib.core.rl_module.marl_module import MultiAgentRLModuleSpec
+
+    spec = MultiAgentRLModuleSpec(
+        marl_module_class=BCTorchMultiAgentModuleWithSharedEncoder,
+        module_specs={
+            "local_2d": SingleAgentRLModuleSpec(
+                observation_space=gym.spaces.Dict({
+                    "global": gym.spaces.Box(low=-1, high=1, shape=(2,)),
+                    "local": gym.spaces.Box(low=-1, high=1, shape=(2,)),
+                }),
+                action_space=gym.spaces.Discrete(2),
+                model_config_dict={"fcnet_hiddens": [64]},
+            ),
+            "local_5d": SingleAgentRLModuleSpec(
+                observation_space=gym.spaces.Dict({
+                    "global": gym.spaces.Box(low=-1, high=1, shape=(2,)),
+                    "local": gym.spaces.Box(low=-1, high=1, shape=(5,)),
+                }),
+                action_space=gym.spaces.Discrete(5),
+                model_config_dict={"fcnet_hiddens": [64]},
+            ),
+        },
+    )
+
+    module = spec.build()
+
+Extending Existing RLlib RL Modules
```

Comment:
```suggestion
Extending Existing RLlib RLModules
```

Reply:
Not sure if there is a distinction between `RL Modules` and `RLModules`. In case there wasn't, I was trying to make the title consistent.